src/EnergyPlus/Pipes.cc

```diff
@@ -134,130 +135,49 @@ namespace Pipes {
 		LocalPipe.deallocate();
 	}
 
-	void
-	SimPipes(
-		int const CompType,
-		std::string & PipeName,
-		int & CompIndex,
-		Real64 const EP_UNUSED( MaxVolFlowRate ),
-		bool const EP_UNUSED( InitLoopEquip ),
-		bool const EP_UNUSED( FirstHVACIteration )
-	)
-	{
-		// SUBROUTINE INFORMATION:
-		//       AUTHOR         Dan Fisher
-		//       DATE WRITTEN   Sept. 1998
-		//       MODIFIED       na
-		//       RE-ENGINEERED  na
-
-		// PURPOSE OF THIS SUBROUTINE:
-
-		// METHODOLOGY EMPLOYED:
-
-		// REFERENCES:
-
-		// Using/Aliasing
-		using InputProcessor::FindItemInList;
-		using InputProcessor::SameString;
-		using General::TrimSigDigits;
-		using DataPlant::PlantLoop;
-		using DataPlant::ScanPlantLoopsForObject;
-		using PlantUtilities::SafeCopyPlantNode;
-		using PlantUtilities::InitComponentNodes;
-		using DataGlobals::BeginEnvrnFlag;
-
-		// Locals
-		// SUBROUTINE PARAMETER DEFINITIONS:
-
-		// SUBROUTINE ARGUMENT DEFINITIONS:
-
-		// INTERFACE BLOCK SPECIFICATIONS
-
-		// DERIVED TYPE DEFINITIONS
-
-		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-		int PipeNum;
-		//  INTEGER :: LoopNum
-		//  INTEGER :: LoopSideNum
-		//  INTEGER :: BranchNum
-		//  INTEGER :: CompNum
-		int FoundOnLoop;
-		bool errFlag;
-
-		//FLOW
-
-		//GET INPUT
+	std::shared_ptr< PlantComponent > LocalPipeData::factory( int objectType, std::string objectName ) {
+		// Process the input data for pipes if it hasn't been done already
 		if ( GetPipeInputFlag ) {
```

Comment:
`GetPipeInputFlag` should really be part of the LocalPipeData class BUT since `LocalPipeData::factory` is static this doesn't work out well. `GetPipeInputFlag` would need to be a static member variable and we would run into the same issue when unit testing (and why there is clear_state now). Honestly, `LocalPipeData::factory` should pass in any data it needs to construct a new object. And that is a SINGLE object it creates, not all objects. 

What if the functionality is:

1) Try to find object in Array
2) If there, return pointer
3) If not, then process passed in input data or call to a (new) InputProcessor function that returns the input data for that specific object
4) Then construct said new object from input data. No grabbing from global data, only function calls.

No need for `GetPipeInputFlag` and `GetPipeInput()` is rolled into the factory basically. Makes unit testing easier since you can pass in or construct elsewhere the input data. Then call `LocalPipeData::factory` as many times as you want to construct or find that specific object.